### DISCUSSION TITLE
Add the option to see all entities in the History tab?

### DISCUSSION THREAD

**Discussion by nedevski** [home-assistant/frontend#13146] [Other feature requests]
Hi, the new History page is awesome and really helps when debugging multiple devices/groups.

However I think it's very useful for debugging to just glance over the full entity list and see if you have any extra ones that you should disable or that are updating way too much, wasting space on the server. I don't want to have all of my entities in a "System" zone and I don't have any other way to see *everything* at once.

On multiple ocasions the history tab was the way that I found I should exclude a certain entity from the recorder because of a lot of useless data. A couple of times I noticed that I accidentally renamed all of my phone's sensors because of a mistake (which broke all of my automations). The new page feels like it has that functionality taken away and I haven't been able to enjoy it because of that.

I'm not asking about full revert to the old page, just a checkbox or a button that will show everything in the system the same way as before. Or just have the page load everything by default and then only filter out things if the user selects something from the new buttons/dropdowns.

**Comment by phippodoplis** [comment]
Even if you do go through the trouble to set up the history page as you wish, when you click on a device in the overview and then click "More history" it takes you to the history page with all the filters changed to match the device. Very frustrating.
I would prefer the history page to show everything, and then have a filter button that enables filtering, after which you can tweak the filters as you wish.

**Reply by fi-sch** [reply]
On the other hand, having History empty at the start is better approach, especially for less powerful hardware.
Anyhow, even a nicely designed "empty state" with a big button in the middle to "Show All Entities" would suffice. It can also be just a "select all" button within' each dropdown, but that wouldn't be good enough UX.

I think that recent changes just threw the baby out with the bathwater.

**Reply by nedevski** [reply]
Good point about the less powerful hardware.
In this case - have a "Show all entities" checkbox that's not checked by default BUT that remembers your choice after refreshing would solve the problem.

**Reply by fi-sch** [reply]
Although your suggestion to have a persistent option to "Show all" tackles the problem well, I think they would want to 'remember' any setup. Therefore checkbox might not necessarily be a perfect solution (UX/UI wise). Note that this will need to co-exist with dropdown selectors they implemented, both ways.

**Reply by nedevski** [reply]
A checkbox was my "brainstorming suggestion". 

I'd say add a 4th button which when pressed will show all entities, when pressed again clears all entities (or the other way around). Have that button state cached/stored in cookies with the rest of the buttons and you should be good to go?

EDIT: Just went over to my HA instance and yes - the state of the 3 buttons persists between page reloads. So adding another one in that group shouldn't be that hard, I hope

**Reply by fi-sch** [reply]
Yeah, you're right. I would go with the button, classic one or a discreet/text one.

BTW, I am pretty disappointed that I was not chosen to their user research group. I am a UX designer with 6+ years of experience and not that small HA setup – 396 sensor entities, 209 native automations, quite a few add-ons, not to mention multiple ESPHome projects.

**Reply by nedevski** [reply]
And I'm getting inceasingly pissed that I'm a .NET dev and HA is written in Python. 😆 Tough love.

I hope I will soon have some spare time to spin up a dev environment, learn a bit of Python and hopefully be able to just submit PRs when something annoys me..

**Comment by 3ative** [comment]
Agreed, Plus, this feature/function would allow displaying only entities that have a "history" set...
- E.G. Listed in "Recorder:" within (configuration.yaml)

_Additional:
As this new History Panel is Per Browser, it would make it a lot easier to set among multiple browser devices, like Tablets / Phone Apps / Etc._

**Reply by xraver** [reply]
+1 :) agree with you. New history is good. but a button to include all entities is realy needed!
thanks

**Comment by fi-sch** [comment]
+1 here

Why?
Current solution requires user to know the entity name/ID he wants to display. In case user doesn't know/remember the name, finding such entity will require more effort and time, then just 'displaying all', scrolling though and "scanning" the page.

My use case:
I have some new entities popping-up each time I set up a new integration. I want to keep it like that. It makes me remember the "new" entities that I "need to take care about" – play a bit more with the integration, adjust it's configuration, and eventually exclude entity from recorder.

**Reply by 3ative** [reply]
In addition, if a user accidentally adds an entity twice, deleting the dup will delete both!!!

**Comment by petrikarj** [comment]
I consider this new history to have a BIG BUG as it lost the ability to show all devices currently rendering the usability to non existent. A quick bug fix like a select all checkbox is needed to bring the usability to on par with earlier version. After the quick fix I hope that the developer considers more advanced method like use of wildcards for filtering…

**Comment by 3ative** [comment]
Here's another bug...
For some reason my chosen entities all disappeared !!! 

- I had to add them all again 😭 and try to remember in what order too!
- Such a pain 

![image](https://user-images.githubusercontent.com/51385971/179778465-72516354-fdaf-460a-aa5e-48717b6ad731.png)

**Reply by zsarnett** [reply]
If you use particular setups consistently create a Dashboard with those entities.

**Reply by fi-sch** [reply]
That's a separate problem and FR I suppose – To be able to save a query/preset based on current selection and restore it on demand.

**Reply by zsarnett** [reply]
Not necessarily. You can just build it manually as you add them all manually. We do not need that as a feature

**Reply by fi-sch** [reply]
This is still something different as you want History panel to remember your selection.

**Reply by zsarnett** [reply]
This history panel does remember your selection

**Reply by zsarnett** [reply]
It's cached and also in the URL...

URL overrides the Cache. So if you bookmark certain pages then you can always get back to them

**Reply by fi-sch** [reply]
Oh, ok. I did not get it at a first place then. Sorry for the confusion.

**Reply by 3ative** [reply]
> If you use particular setups consistently create a Dashboard with those entities.

Will this dashboard show lined up On/Off States and those nice full width graphs grouped by type?
E.G. :
![image](https://user-images.githubusercontent.com/51385971/179804494-6ac62ba6-8c41-497d-ad81-8c530f7bd655.png)

**Reply by zsarnett** [reply]
If you make it a 1 card

**Reply by 3ative** [reply]
> If you make it a 1 card

Ah.. so, we can use the same Google Graphs in any Dashboard page... is that correct?

**Reply by zsarnett** [reply]
Yea with the History Graph card

**Reply by 3ative** [reply]
>Yea with the History Graph card

Well, that works... Thanks, good call
![image](https://user-images.githubusercontent.com/51385971/179842558-5ee72051-2376-4dce-9a68-97f128f86a30.png)

Next question, what about these type bars, any equivalent ? 

![image](https://user-images.githubusercontent.com/51385971/179842747-58a56aae-914a-4bdd-86ab-6231c489655a.png)

**Reply by zsarnett** [reply]
If you add those entities to the card then it will work the same way

**Reply by 3ative** [reply]
> If you add those entities to the card then it will work the same way

Sweet, a little annoying (I hate work-arounds) but at least I have a clean page of all the entities I track now. 

I made a new empty dashboard, so it shows in the left menu.

![Screenshot 2022-07-19 23 56 16](https://user-images.githubusercontent.com/51385971/179862110-cc81c472-7533-476c-b411-e2667dafc465.png)


Thanks for the support and taking the time to respond...

I'd still prefer the 'Old History' page, coupled with Recorder 'filter' back.

**Comment by 3ative** [comment]
And we've not even mentioned the impact of having all those entities applied to the Mobile app...
I have to scroll down two full pages before I even get to the first Graph(s)

**Reply by fi-sch** [reply]
Good point. History on mobile is pretty useless.
Not to mention that (even on desktop) you can't do a multiselect and need to open a dropdown each time you want to display a new entity.
Also, why would they show in those dropdowns an entities that doesn't have any history, cause they are excluded from recorder? That makes selection even harder.

**Reply by 3ative** [reply]
> Also, why would they show in those dropdowns an entities that doesn't have any history, cause they are excluded from recorder? That makes selection even harder.

Indeed. For those that took the time to set everything in recorder... so they don't show in History, This new one is V/Confusing and a massive pain to select from that huge 'every thing' drop-down list.

**Reply by 3ative** [reply]
What would be good is, if Recorder had its own UI panel where we could set the entities we wanted shown AND the order for the History Page/Panel to respect.

**Reply by Misiu** [reply]
There is a draft PR with recorder config panel: https://github.com/home-assistant/frontend/pull/11002, but sadly core PR (https://github.com/home-assistant/core/pull/62604) got closed :(

**Comment by 3ative** [comment]
Also...
Good to see a DEV (@zsarnett) on this thread - Nice to know they are here too.  👍

**Comment by lsdlsd88** [comment]
any update????? why would you break a previously working option??

**Comment by 3ative** [comment]
Here's a 'Work-a-round' tutorial I made:
https://youtu.be/8XivwDe6DNY

**Reply by fi-sch** [reply]
Great idea, thanks! I guess that if used with [lovelace-auto-entities](https://github.com/thomasloven/lovelace-auto-entities), this could solve the problem with auto-populating the History.

**Reply by phippodoplis** [reply]
The thing missing from these suggestions is the date selection that you can do in the standard history. I use that quite a lot for zooming in on interesting times.

**Comment by phippodoplis** [comment]
Since it seems to be fairly easy for people who want to tune the history view to create a new view with a history card, can we revert the history view to the original?

**Reply by zsarnett** [reply]
It will not be reverted

**Reply by phippodoplis** [reply]
So, can you tell us what is going to happen? All I'm getting at the moment is the original issue closed with no discussion, and now this discussion which seems to be going nowhere. Just a quick response to reassure us it will get better is all I need.

**Reply by zsarnett** [reply]
I have discussed in other places I guess but not here. We will add some button or control that will allow adding all when we figure out the best UI for this

**Reply by fi-sch** [reply]
That's great to hear, thanks!

**Reply by darmach** [reply]
@zsarnett Thanks, that would be great. Losing that all-entities history page made me realise how often I glimpsed over it :)

**Reply by 3ative** [reply]
@darmach 
In the mean-time, maybe this could be a 'work-around' option ?
https://youtu.be/8XivwDe6DNY

**Comment by fi-sch** [comment]
Any updates here?
This seems to be forgotten, so I've created a relevant WTH topic: https://community.home-assistant.io/t/wth-cant-i-see-all-entities-from-recorder-history-in-history-view/468454
